org.eclipse.scanning.test/src/org/eclipse/scanning/test/command/AbstractMScanTest.java

```diff
@@ -0,0 +1,432 @@
+		    {@literal    <property name="submitQueue" value="uk.ac.diamond.p45.submitQueue" />}
+		    {@literal    <property name="statusSet"   value="uk.ac.diamond.p45.statusSet"   />}
+		    {@literal    <property name="statusTopic" value="uk.ac.diamond.p45.statusTopic" />}
+		    {@literal </bean>}
+
+		 */
+		servlet = new ScanServlet();
+		servlet.setBroker(uri.toString());
+		servlet.setDurable(true);
+		servlet.connect(); // Gets called by Spring automatically
+		
+	}
+
+	@AfterClass
+	public static void disconnect()  throws Exception {
+		servlet.getConsumer().cleanQueue(servlet.getSubmitQueue());
+		servlet.getConsumer().cleanQueue(servlet.getStatusSet());
+		servlet.disconnect();
+	}
+
+	protected String path;
+	private boolean requireFile;
+	
+	public AbstractMScanTest(boolean requireFile) {
+		this.requireFile = requireFile;
+	}
+	
+	@Before
+	public void before() throws Exception {
+		
+		if (requireFile) {
+			File output = File.createTempFile("test_nexus", ".nxs");
+			output.deleteOnExit();
+			path = output.getAbsolutePath().replace("\\\\", "\\").replace('\\', '/');
+		}
+		
+		servlet.getConsumer().cleanQueue(servlet.getSubmitQueue());
+		servlet.getConsumer().cleanQueue(servlet.getStatusSet());
+		
+		if (pconsumer!=null) {
+			pconsumer.cleanQueue(PROCESSING_QUEUE_NAME);
+			pconsumer.cleanQueue("scisoft.operation.STATUS_SET");
+		}
+	}
+
+	protected List<ScanBean> runAndCheck(String name, boolean blocking, long maxScanTimeS) throws Exception {
+		
+		final IEventService eservice = Services.getEventService();
+
+		// Let's listen to the scan and see if things happen when we run it
+		final ISubscriber<IScanListener> subscriber = eservice.createSubscriber(new URI(servlet.getBroker()), servlet.getStatusTopic());
+		final ISubmitter<ScanBean>       submitter  = eservice.createSubmitter(new URI(servlet.getBroker()),  servlet.getSubmitQueue());
+		
+		try {
+			final List<ScanBean> beans = new ArrayList<>(13);
+			final List<ScanBean> failed = new ArrayList<>(13);
+			final List<ScanBean> startEvents = new ArrayList<>(13);
+			
+			final CountDownLatch latch = new CountDownLatch(1);
+			subscriber.addListener(new IScanListener() {
+				@Override
+				public void scanEventPerformed(ScanEvent evt) {
+					if (evt.getBean().getStatus()==Status.FAILED) failed.add(evt.getBean());
+					if (evt.getBean().getPosition()!=null) {
+						beans.add(evt.getBean());
+					}
+				}
+	
+				@Override
+				public void scanStateChanged(ScanEvent evt) {
+					if (evt.getBean().getStatus()==Status.FAILED) failed.add(evt.getBean());
+					if (evt.getBean().scanStart()) {
+						startEvents.add(evt.getBean()); // Should be just one
+					}
+	                if (evt.getBean().scanEnd()) {
+	                	latch.countDown();
+	                }
+				}
+			});
+	
+			
+			// Ok done that, now we sent it off...
+			pi.exec("submit("+name+", block="+(blocking?"True":"False")+", broker_uri='"+uri+"')");
+			
+			Thread.sleep(200);

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)